src/components/learner-credit-management/cards/NewAssignmentModalButton.jsx

```diff
@@ -84,32 +124,63 @@ const NewAssignmentModalButton = ({ course, children }) => {
           httpErrorStatus,
           httpErrorResponseData,
         } = err.customAttributes;
+        let locallyScopedErrorReason = 'system_error';
         if (httpErrorStatus === 422) {
           const responseData = JSON.parse(httpErrorResponseData);
           setCreateAssignmentsErrorReason(responseData[0].reason);
+          locallyScopedErrorReason = responseData[0].reason;
         } else {
           setCreateAssignmentsErrorReason('system_error');
```

Comment:
nit: I'm wondering if this line should rely on the `locallyScopedErrorReason` such that the string constant `system_error` doesn't need to be repeated twice in this function.

Might also consider renaming `locallyScopedErrorReason` to just `errorReason`, e.g.:

```jsx
let errorReason = 'system_error';
if (httpErrorStatus === 422) {
  const responseData = JSON.parse(httpErrorResponseData);
  errorReason = responseData[0].reason;
  setCreateAssignmentsErrorReason(errorReason);
} else {
  setCreateAssignmentsErrorReason(errorReason);
}
setAssignButtonState('error');
sendEnterpriseTrackEvent(...);
```